upup/pkg/fi/cloudup/scalewaytasks/loadbalancer.go

```diff
@@ -0,0 +1,289 @@
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// +kops:fitask
+type LoadBalancer struct {
+	Name      *string
+	Lifecycle fi.Lifecycle
+
+	Region       *string
+	LBID         *string
+	LBAddresses  []string
+	Tags         []string
+	ForAPIServer bool
+}
+
+var _ fi.CompareWithID = &LoadBalancer{}
+var _ fi.HasAddress = &LoadBalancer{}
+
+func (l *LoadBalancer) CompareWithID() *string {
+	return l.LBID
+}
+
+func (l *LoadBalancer) IsForAPIServer() bool {
+	return l.ForAPIServer
+}
+
+func (l *LoadBalancer) Find(context *fi.CloudupContext) (*LoadBalancer, error) {
+	if fi.ValueOf(l.LBID) == "" {
+		return nil, nil
+	}
+
+	cloud := context.T.Cloud.(scaleway.ScwCloud)
+	lbService := cloud.LBService()
+
+	loadBalancer, err := lbService.GetLB(&lb.GetLBRequest{
+		Region: scw.Region(cloud.Region()),
+		LBID:   fi.ValueOf(l.LBID),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("getting load-balancer %s: %s", fi.ValueOf(l.LBID), err)
+	}
+
+	lbIPs := []string(nil)
+	for _, IP := range loadBalancer.IP {
+		lbIPs = append(lbIPs, IP.IPAddress)
+	}
+
+	return &LoadBalancer{
+		Name:         &loadBalancer.Name,
+		LBID:         &loadBalancer.ID,
+		LBAddresses:  lbIPs,
+		Tags:         loadBalancer.Tags,
+		Lifecycle:    l.Lifecycle,
+		ForAPIServer: l.ForAPIServer,
+	}, nil
+}
+
+func (l *LoadBalancer) FindAddresses(context *fi.CloudupContext) ([]string, error) {
+	cloud := context.T.Cloud.(scaleway.ScwCloud)
+	lbService := cloud.LBService()
+
+	if l.LBID == nil {
+		return nil, nil
+	}
+
+	loadBalancer, err := lbService.GetLB(&lb.GetLBRequest{
+		Region: scw.Region(cloud.Region()),
+		LBID:   fi.ValueOf(l.LBID),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	addresses := []string(nil)
+	for _, address := range loadBalancer.IP {
+		addresses = append(addresses, address.IPAddress)
+	}
+
+	return addresses, nil
+}
+
+func (l *LoadBalancer) Run(context *fi.CloudupContext) error {
+	return fi.CloudupDefaultDeltaRunMethod(l, context)
+}
+
+func (_ *LoadBalancer) CheckChanges(actual, expected, changes *LoadBalancer) error {
+	if actual != nil {
+		if changes.Name != nil {
+			return fi.CannotChangeField("Name")
+		}
+		if changes.LBID != nil {
+			return fi.CannotChangeField("ID")
+		}
+		if changes.Region != nil {
+			return fi.CannotChangeField("Region")
+		}
+	} else {
+		if expected.Name == nil {
+			return fi.RequiredField("Name")
+		}
+		if expected.Region == nil {
+			return fi.RequiredField("Region")
+		}
+	}
+	return nil
+}
+
+func (l *LoadBalancer) RenderScw(t *scaleway.ScwAPITarget, actual, expected, changes *LoadBalancer) error {
+	lbService := t.Cloud.LBService()
+	region := scw.Region(fi.ValueOf(expected.Region))
+	var loadBalancer *lb.LB
+	backEndToCreate := true
+	frontEndToCreate := true
+
+	if actual != nil {
+		klog.Infof("Updating existing load-balancer with name %q", expected.Name)
+
+		lbToUpdate, err := lbService.GetLB(&lb.GetLBRequest{
+			Region: region,
+			LBID:   fi.ValueOf(actual.LBID),
+		})
```

Comment:
Probably you should not get the LB here again. It's already found in Find() and you should have the required fields set. You could add `Description` and `SslCompatibilityLevel` as part of the LB struct and pass them in model to the desired value.